src/test/java/org/minions/devfund/chelo/SquarelotronTest.java

```diff
@@ -0,0 +1,227 @@
+import org.junit.Test;
+
+import static org.junit.Assert.assertNotEquals;
+import static org.junit.Assert.assertEquals;
+
+/**
+ * Test Class Squarelotron.
+ */
+public class SquarelotronTest {
+
+    /**
+     * A test to validate the Main diagonal flip dimension two.
+     */
+    @Test
+    public void testMainDiagonalFlipDimensionTwo() {
+        final int size = 2;
+        final int ring = 1;
+        final int[][] expectedArray = new int[][]{
+                {3, 4},
+                {1, 2}};
+
+        Squarelotron squarelotron = new Squarelotron(size);
+        assertEquals(expectedArray, squarelotron.upsideDownFlip(ring).squarelotron);
+    }
+
+    /**
+     * A test to validate the Main diagonal flip dimension five ring three.
+     */
+    @Test
+    public void testMainDiagonalFlipDimensionFiveRingThree() {
+        final int size = 5;
+        final int ring = 3;
+        final int[][] expectedArray = new int[][] {
+                {1, 2, 3, 4, 5},
+                {6, 7, 8, 9, 10},
+                {11, 12, 13, 14, 15},
+                {16, 17, 18, 19, 20},
+                {21, 22, 23, 24, 25}};
+
+        Squarelotron squarelotron = new Squarelotron(size);
+        assertEquals(expectedArray, squarelotron.mainDiagonalFlip(ring).squarelotron);
+    }
+
+    /**
+     * A test to validate the Main diagonal flip dimension five ring two.
+     */
+    @Test
+    public void testMainDiagonalFlipDimensionFiveRingTwo() {
+        final int size = 5;
+        final int ring = 2;
+        final int[][] expectedArray = new int[][]{
+                {1, 2, 3, 4, 5},
+                {6, 7, 12, 17, 10},
+                {11, 8, 13, 18, 15},
+                {16, 9, 14, 19, 20},
+                {21, 22, 23, 24, 25}};
+
+        Squarelotron squarelotron = new Squarelotron(size);
+        assertEquals(expectedArray, squarelotron.mainDiagonalFlip(ring).squarelotron);
+    }
+
+    /**
+     * A test to validate the Main diagonal flip dimension five ring one.
+     */
+    @Test
+    public void testMainDiagonalFlipDimensionFiveRingOne() {
+        final int size = 5;
+        final int ring = 1;
+        final int[][] expectedArray = new int[][]{
+                {1, 6, 11, 16, 21},
+                {2, 7, 8, 9, 22},
+                {3, 12, 13, 14, 23},
+                {4, 17, 18, 19, 24},
+                {5, 10, 15, 20, 25}};
+
+        Squarelotron squarelotron = new Squarelotron(size);
+        assertEquals(expectedArray, squarelotron.mainDiagonalFlip(ring).squarelotron);
+    }
+
+    /**
+     * A test to validate the upside down flip.
+     */
+    @Test
+    public void testUpsideDownFlip() {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Update this method so that its implementation is not identical to "testMainDiagonalFlipDimensionTwo" on line 15. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS4144)